src/main/scala/in/ashwanthkumar/vamana2/apps/HadoopScalar.scala

```diff
@@ -30,12 +30,13 @@ class HadoopScalar extends Scalar[HDemand, HSupply] {
   override def requiredNodes(demand: HDemand, supply: HSupply, ctx: Context): Int = {
     log.info(s"Demand found is $demand")
     log.info(s"Supply found is $supply")
-    // If the demand is nothing, scale down to cluster min size
-    // If the cluster is running with min capacity and demand > supply, scale it up to max size
-    // else keep the cluster intact
-    if (demand.quantity == 0.0) ctx.cluster.minNodes
-    else if (demand.quantity > supply.available && ctx.currentSize < ctx.cluster.maxNodes) ctx.cluster.maxNodes
-    else ctx.currentSize
+
+    if (demand.quantity == 0.0)
+      return ctx.cluster.minNodes
```

Comment:
@manojlds I prefer not to use `return` in scala[1]. Can we re-write this without using it? 

[1] - https://tpolecat.github.io/2014/05/09/return.html